certora/specs/HealthStatusInvariant.spec

```diff
@@ -0,0 +1,207 @@
+*  and some authentication is done
+* - After this, callWithContextInternal invokes the transfer function 
+* on the collateral address
+* - Collaterals in the EVK must all be Token.sol and token's transfer
+* implementation calls initOperation which enqueues an account status
+* check on the EVC for the onBehalfOfAccount it also gets from EVC.
+* Because onBehalfOfAccount was set to the from address in callWithContextInternal this status check is for the from account
+* To emulate this, we:
+* - explicitly call EToken.transferFrom using the expected addresses
+* - enqueue a status check on the evc for the "from" address
+*/
+function CVLEnforceCollateralTransfer(env e, address collateral, uint256 amount, address from, address receiver) {
+    // (Ideally: Cast collateral address into EToken to allow for
+    // multiple addresses with the EToken contract)
+    evc.requireAccountStatusCheck(e, from);
+    EToken.transferFromInternalHarnessed(e, from, receiver, amount);
+}
+
+// Assuming the prices stay the same, a healthy account can never become 
+// unhealthy. Here, our assumption that the prices do not change is implicit
+// in the fact that the summary for GetQuote is an uninterpreted function --
+// the prover will model it as a function so it will always return the same
+// value when given the same arguments.
+rule accountsStayHealthy (method f) filtered { f -> 
+    // Literal selectors are used to avoid compilation errors when
+    // only some of the modules are in the verification scene
+    // sig:GovernanceModule.clearLTV(address).selector
+    f.selector != 0x8255d029 && 
+    // sig:GovernanceModule.setLTV(address,uint16,uint16,uint32).selector
+    f.selector != 0x4bca3d5b &&
+    // sig:InitializeModule.initialize(address).selector
+    f.selector != 0xc4d66de8 
+}{
+    env e;
+    calldataarg args;
+    address account;
+    address[] collaterals = evc.getCollaterals(e, account);
+    require collaterals.length == 2; // loop bound
+    require oracleAddress != 0;
+
+    // Vault cannot be a user of itself
+    require account != currentContract;
+    // Vault should not be used as a collateral
+    require collaterals[0] != currentContract;
+    require collaterals[1] != currentContract;
+    // Collaterals must be ETokens
+    require collaterals[0] == EToken;
+    require collaterals[1] == EToken;
+    // not sure the following 4 are really needed
+    require account != erc20;
+    require account != oracleAddress;
+    require account != evc;
+    require account != unitOfAccount;
+    
+    require LTVConfigAssumptions(e, getLTVConfig(e, collaterals[0]));
+    require LTVConfigAssumptions(e, getLTVConfig(e, collaterals[1]));
+
+    // Otherwise this can cause an unintersting divide by zero in OwedLib.getCurrentOwed (on the mulDiv)
+    require getUserInterestAccumulator(e, account) > 0;
+    require storage_interestAccumulator(e) == getUserInterestAccumulator(e, account);
```

Comment:
this is assuming interest has already been accumulated right?  better to comment on this, and do we have a rule that shows that interest is always accumulated ?

Reply:
I wound up detelting `accountsStayHealthy` since `accountsStayHealthy_strategy` is the one that worked for all of the methods. the _strategy version does not have this assumption, so essentially this is resolved. Thanks for raising this!